File: ds12_NoIntro.py
msg='Place in game folder and launch this script.'
# supports both games
import os

n1='Dead Space.exe'
s1=b'XCENTKOWSK_C8A99CD_622DBBB_v3',b'XCENTKOWSK_C78C369_F71988A_v3'

n2='deadspace2.exe'
s2=b'XCENTKOWSK_C9C50BC_08B592A_v3',b'XPATEL1026_CB19946_663E415_v3'

d = 300 # step

x='The End.'

# os.chdir('..')

for i in 1,2:
        
    if not os.path.isfile(n1):
        n1,s1=n2,s2
        if i==2:
            x=msg
        continue
    else:
        print(n1)

    # s1=s1[::-1] # revert changes
            
    f=open(n1,'rb+')
    f.seek(0)
    fd=1
    while fd:
        fd=f.read(d)
        k = fd.count(s1[0])
        if k:
            print('patching')
            fr=fd.replace(*s1)
            f.seek(-d,1)
            f.write(fr)
            break
        
    f.close()
    
    if not k:
        print('was not changed')
    
input(x)

# step is number of bytes red at time
# to be changed in case string didn't fit completely in buffer

File: ds12-nointro.lua
pet='Press Enter to '
print(pet..'start.')
io.read()
s1={"Dead Space.exe",
'XCENTKOWSK_C8A99CD_622DBBB_v3',
'XCENTKOWSK_C78C369_F71988A_v3'}
s2={"deadspace2.exe",
'XCENTKOWSK_C9C50BC_08B592A_v3',
'XPATEL1026_CB19946_663E415_v3'}

for i=1,2 do
print(s1[1])
f = io.open(s1[1], "r+")
if f then break else print('not found.') s1=s2 end
end

if not f then 
print('Place in game folder and try again.') 
goto quit
end

msg = 'was not found.'

block = 300
  while true do
x = f:seek()
b1 = f:read(block)
if not b1 then break; msg = 'quit' end
if b1:find(s1[3]) then msg = 'is already removed.' break end
i,j = b1:find(s1[2])
   if i then 
f:seek("set",x+i-1)
f:write(s1[3])
msg = 'eliminated.'
break
   end
  end

f:close()
print('Intro '..msg)
::quit::
print(pet..'quit.')
io.read()
